deploy/org.mavlink.qgroundcontrol.metainfo.xml

```diff
@@ -24,6 +24,10 @@
 
   <content_rating type="oars-1.1" />
 
+  <releases>
+    <release date="2023-03-21" version="4.2.6"/>
```

Comment:
this tag and date should be generated by the build system over using it manually

Reply:
Is there a reliable way to extract the version? I don't see it in the `CMakeLists.txt`. `git` is not an option as someone may build the source from an archive.

Reply:
> Is there a reliable way to extract the version? I don't see it in the `CMakeLists.txt`. `git` is not an option as someone may build the source from an archive.

It's not possible to build from archive since we use submodules already.

Reply:
Can't you distribute the main source and the source of the dependencies as an archive? I.e. checkout the repo with submodules, delete the `.git` folder and then zip all the files.

How would I get the version in this case? How is it set for the compiled executable?

Reply:
GitHub archives in releases does not contain submodules.
Deleting `.git` is not an use case, the file will be updated by the build system on CI with release tags.

Reply:
Can you give me a pointer where the release version is extracted during CI?